ClockifyAPI.py

```diff
@@ -450,8 +644,24 @@ def addEntry(self, start, description, projectName, userMail, workspace,
             url = self.url + "/workspaces/%s/time-entries"%wsId
             
             if projectName != None:
-                projectId = self.getProjectID(projectName, workspace)
+                if (self._syncProjects == True):
+                    projectId = self.getProjectID(projectName, workspace, skipPrjQuery=True)
+                    if taskName != None:
+                        pTasks = self.getTasksOnProject(workspace, projectName)
+                        taskId = self.getTaskIdFromTasks(taskName, pTasks)                   
+                        self.logger.info("Found task %s in project %s"%(taskName, projectName))
+                    else:
+                        taskId = None
+                else:
+                    projectId = self.getProjectID(projectName, workspace)
+                    if taskName != None:
+                        pTasks = self.getTasksOnProject(workspace, projectName)
+                        taskId = self.getTaskIdFromTasks(taskName, pTasks)
+                        self.logger.info("Found task %s in project %s"%(taskName, projectName))                   
+                    else:
+                        taskId = None
```

Comment:
Yeah, that's something I wasn't sure of. What I had in mind there - but I'm still not sure that it is what's happening is something like:

>  if a taskName exists, 
>    and if Projects have already been queried, then do the following without querying all the projects again 
>    or else query the Projects
> etc. 

I.e. I was trying to avoid the script to check if the projects have already been loaded before. But I can't say if actually serves the purpose.

Reply:
Project queries add quite a long time to the process, hence trying to optimize it.

Reply:
Yeah sure, that's a good idea! I think ig you add `skipPrjQuery=self._syncProjects` to the arguments list of self.getProjectID than you have the desired behaviour, or am I missing something?